### PR TITLE
Fix issues related to exception handling while unplugging the board.

### PR DESCRIPTION
**Pull request by AlexandraTrifan**
- Unplugging the board during the calibration should not crash the app, even if the calibration is throwing exceptions. We handle those and disconnect without errors.
- Unplugging the board during the tool construction should not crash the app, even if the constructors are throwing exceptions. We handle those in the Tool Launcher and disconnect without errors.
- We should not suppose that the calibration and connection were done well and all the tools are available from that point on.

- Make sure the Power Supplies are not ON when first connecting to a board.
- Don't use averaging in the "Sample" Type for Spectrum Analyzer.

### CODE REVIEW

File: src/tool_launcher.hpp [hunk 1]
```diff
@@ -260,6 +260,8 @@ private Q_SLOTS:
 
 	bool m_useNativeDialogs;
 
+	bool m_dac_tools_failed, m_adc_tools_failed;
```
Comment (DanielGuramulta) [inline src/tool_launcher.hpp:263]: are this values initialised in the constructor of the ToolLauncher?

File: src/toolmenuitem.cpp [hunk 2]
```diff
@@ -121,6 +121,15 @@ bool ToolMenuItem::eventFilter(QObject *watched, QEvent *event)
 	return QObject::event(event);
 }
 
+void ToolMenuItem::setCalibrating(bool calibrating)
```
Comment (DanielGuramulta) [inline src/toolmenuitem.cpp:124]: 👍

**Review by DanielGuramulta — APPROVED** [review]
Looks good!

File: src/tool_launcher.hpp (hunk 1 above)
Comment (AlexandraTrifan) [reply src/tool_launcher.hpp:263]: I added the initialization now.

**Pull request merged**